test/e2e/settings/disable_catch_all.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package settings
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	extensions "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+const (
+	waitForLuaSync = 5 * time.Second
+)
+
+var _ = framework.IngressNginxDescribe("Disabled catch-all", func() {
+	f := framework.NewDefaultFramework("disabled-catch-all")
+
+	BeforeEach(func() {
+		f.NewEchoDeployment()
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should ignore catch all Ingress", func() {
+
+		name := "foo"
+		ingKey := fmt.Sprintf("%v/%v", f.IngressController.Namespace, name)
```

Comment:
can you not use the existing helper function `k8s.MetaNamespaceKey` for this?

Reply:
We have no object that implements `meta.Interface` at this point.